pkg/codegen/java/gen.go

```diff
@@ -490,7 +490,7 @@ func (pt *plainType) genInputProperty(ctx *classFileContext, prop *schema.Proper
 	return nil
 }
 
-func (pt *plainType) genInputType(ctx *classFileContext) error {
+func (pt *plainType) genInputType(ctx *classFileContext, propertiesAsOutputs bool) error {
```

Comment:
Probably, maybe. Ideally it would be localized to where it is actually used, but that is probably a bigger refactoring and there's no point in making it part of this PR.

The least damage we can do to an already difficult to maintain codebase is probably our best bet at this moment.

I'm sorry I don't have a more constructive comment, maybe others will have a better idea.

Reply:
I've looked at the code again and I don't have an idea for a constructive comment with a specific action,
so feel free to ignore me. Sorry about that.